//------------------------------------------------------------------------------
// protobuf specification for network agent services
// all network services are captured in this file
//------------------------------------------------------------------------------

syntax = "proto3";

import "types.proto";

package net;

// The Network service definition
service Net {
  rpc VlanCreate (VlanRequest) returns (VlanResponse) {}
  rpc VlanUpdate (VlanRequest) returns (VlanResponse) {}
}

// VlanRequest message is used to do add/mod/del/get of vlan
message VlanRequest {
  types.L2SegmentType    type = 1;
  uint32                 vlan_id= 2;
  types.EncapInfo        access_encap = 3;
  types.EncapInfo        fabric_encap = 4;
}

// The response message contains the status
message VlanResponse {
  uint32    status = 1;
}
